analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/ObjectCreation.cs

```diff
@@ -0,0 +1,52 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors;
+
+internal sealed class ObjectCreation : SimpleProcessor<IObjectCreationOperationWrapper>
+{
+    protected override IObjectCreationOperationWrapper Convert(IOperation operation) =>
+        IObjectCreationOperationWrapper.FromOperation(operation);
+
+    protected override ProgramState Process(SymbolicContext context, IObjectCreationOperationWrapper operation)
+    {
+        if (operation.Type.IsNullableValueType())
+        {
+            if (operation.Arguments.IsEmpty)
+            {
+                return context.SetOperationConstraint(ObjectConstraint.Null);
+            }
+            else if (context.State[operation.Arguments.First().ToArgument().Value] is { } value)
+            {
+                return context.State.SetOperationValue(context.Operation, value.WithConstraint(ObjectConstraint.NotNull));
```

Comment:
I don't think we need this with my work. The argument must be a non-nullable value type. Before it gets passed to the argument, there is a reference operation. We will learn on that operation the operation value "NotNull". This is more precise as it also supports e.g. `new int?(true ? a : 42)` as `a` learns "NotNull" on the reference operation and 42 learns NotNull in "ConstantCheck".

Reply:
`SetOperationValue` throws away whatever constraint was there from PreProcess. 
We need it this way to propage other random constraints (like cryptography, or the future CBDE replacement constraints)

So those 2 do not collide (too much). In theory, the value that will arrive should already have your constraint. And in that case, WithConstraint will do NOOP as I added the check there.